src/python/pants/backend/java/dependency_inference/java_parser_test.py

```diff
@@ -111,7 +114,7 @@ class Foo {}
         [
             SourceFilesRequest(
                 (target.get(Sources),),
-                for_sources_types=(JavaSources,),
+                for_sources_types=(JavaSourcesGeneratorSourcesField,),
```

Comment:
It'd be better to use `JavaSourceField` here. If https://github.com/pantsbuild/pants/issues/12953 happens, then `JavaSourcesGeneratorSourcesField` would no longer subclass `Sources` and wouldn't work here.